src/types/__tests__/effector/sample.test.js

```diff
@@ -946,3 +946,40 @@ describe('without clock', () => {
     `)
   })
 })
+
+describe('without fn', () => {
+  test('source Unit, clock Unit, target Event with opt fn', () => {
+    const source = createStore({name: 'John'})
+    const target = createEvent<string>()
+    const clock = createEvent<string>()
+
+    sample({
+      source,
+      clock,
+      target,
+    })
+
+    expect(typecheck).toMatchInlineSnapshot(`
+      "
+      --typescript--
+      No overload matches this call.
```

Comment:
Working overload means that this test should pass, please improve this.

Note that `A extends Combinable` means that such overload is for `source: {foo, bar}`, what imply irrelevant test: it not checking what is really changed